fcli-fod/src/main/java/com/fortify/cli/fod/app/cli/cmd/FoDApplicationCreateCommand.java

```diff
@@ -1,9 +1,147 @@
+    private ArrayList<String> userGroups;
+    @Option(names = {"--microservice", "-ms"}, arity = "0..*", descriptionKey = "microservice")
+    private ArrayList<String> microservices;
+    @Option(names = {"--release-microservice", "-rms"}, descriptionKey = "releaseMs")
+    private String releaseMicroservice;
+
+    @Mixin
+    private FoDAppTypeMixin.AppTypeOption typeMixin;
+    @Mixin
+    private FoDCriticalityTypeMixin.CriticalityTypeOption criticalityMixin;
+    @Mixin
+    private FoDAttributeUpdateMixin.OptionalAttrOption attrUpdateMixin;
+    @Mixin
+    private FoDSdlcStatusTypeMixin.SdlcStatusTypeOption sdlcStatusMixin;
+
+    @SneakyThrows
+    @Override
+    protected Void run(UnirestInstance unirest) {
+        validate();
+
+        ObjectNode body = getObjectMapper().createObjectNode();
+        body.put("applicationName", applicationName)
+                .put("applicationDescription", description == null ? "" : description)
+                .put("businessCriticalityType", String.valueOf(criticalityMixin.getCriticalityType()))
+                .put("emailList", getEmailList(notifications))
+                .put("releaseName", releaseName)
+                .put("releaseDescription", releaseDescription == null ? "" : releaseDescription)
+                .put("sdlcStatusType", String.valueOf(sdlcStatusMixin.getSdlcStatusType()));
+
+        // look up username if supplied to get owner id
+        FoDUserDescriptor userDescriptor = FoDUserHelper.getUser(unirest, owner, true);
+        body.put("ownerId", userDescriptor.getId());
+
+        FoDAppTypeMixin.FoDAppType appType = typeMixin.getAppType();
+        switch (appType) {
+            case Web:
+                body.put("applicationType", "Web_Thick_Client");
+                body.put("hasMicroservices", false);
+                break;
+            case Mobile:
+                body.put("applicationType", "Mobile");
+                body.put("hasMicroservices", false);
+                break;
+            case Microservice:
+                body.put("hasMicroservices", true);
+                body.set("microservices", getMicroservicesNode(microservices));
+                body.put("releaseMicroserviceName", releaseMicroservice == null ? "" : releaseMicroservice);
+                break;
+            default:
+        }
+        body.set("attributes", getAttributesNode(unirest));
+        body.set("userGroupIds", getUserGroupsNode(unirest, userGroups));
+
+        //System.out.println(body.toPrettyString());
+
+        JsonNode postResponse = unirest.post(FoDUrls.APPLICATIONS).body(body).asObject(JsonNode.class).getBody();
+        // retrieve the updated application
+        JsonNode getResponse = unirest.get(FoDUrls.APPLICATION)
+                .routeParam("appId", postResponse.get("applicationId").asText())
+                .asObject(JsonNode.class).getBody();
+        getOutputMixin().write(getResponse);
+        return null;
+    }
+
+    private void validate() {
```

Comment:
I can see how this can be done for mutually exclusive options but what about on the "value" of an option, i.e. if an option has value=x then these two fields are required, value=y then two other fields ?